overlord/hookstate/hookmgr.go

```diff
@@ -58,6 +58,9 @@ type Handler interface {
 
 	// Error is called if the hook encounters an error while running.
 	Error(err error) error
+
+	// SetConf associates the value with the key in the snap's configuration.
+	SetConf(key string, value interface{})
```

Comment:
It doesn't make sense to have this here, as it means every single handler will need to reimplement exactly the same thing for handling the call of a single command of snapctl. Imagine how this will evolve once we have 20 other ctlcmds besides get and set.

Instead, the Handler should remain responsible for activities that the _handler_ needs to be aware of. This confusion is only happening because we chose `apply-config` for our first hook. This is a tricky one because precisely the `get` and `set` functionality is something we want every single hook to support. Other hooks will more clearly distinguish what is appropriate for the handler vs. what is appropriate for the command to do.